### PR TITLE
Implemented Search and user page

### PR DESCRIPTION
**Pull request by Nahlam4**
Implemented Search and user Page with full funcionality

### CODE REVIEW

File: src/main/java/de/tinf22b6/dhbwhub/security/WebSecurityConfig.java [hunk 1]
```diff
@@ -36,16 +36,22 @@ public class WebSecurityConfig {
             "/post/homepage-preview-posts/{id:\\d+}",
             "/post/post-thread/{id:\\d+}",
             "/post/popular-tags",
-            "/post/posts-by-tag/{tag}",
+            "/post/posts-by-tag/*",
             "/post/user-posts/{id:\\d+}",
-            "/post/posts-by-keyword/{keyword}",
-            "/post/posts-by-tag-keyword/{keyword}",
+            "/post/friend-posts/{id:\\d+}",
```
Comment (maxschwinghammer) [inline src/main/java/de/tinf22b6/dhbwhub/security/WebSecurityConfig.java:41]: `"/post/friend-posts/{id:\\d+}"` should not be public, I just deleted this from this config

File: src/main/java/de/tinf22b6/dhbwhub/security/WebSecurityConfig.java [hunk 2]
```diff
@@ -36,16 +36,22 @@ public class WebSecurityConfig {
...
+            "/post/posts-by-keyword/*",
+            "/post/posts-by-tag-keyword/*",
+            "/post/friend-posts-by-tag",
```
Comment (maxschwinghammer) [inline src/main/java/de/tinf22b6/dhbwhub/security/WebSecurityConfig.java:44]: `"/post/friend-posts-by-tag"` should not be public, I just deleted this from this config

File: src/main/web/src/scenes/Search/components/Post.tsx
Comment (maxschwinghammer) [inline src/main/web/src/scenes/Search/components/Post.tsx:1]: Why do you create a new Post component? Please reuse the existing one.

File: src/main/web/src/scenes/Search/components/PostModel.tsx
Comment (maxschwinghammer) [inline src/main/web/src/scenes/Search/components/PostModel.tsx:1]: Please reuse the existing model

File: src/main/web/src/scenes/Search/components/SearchService.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+import {useDetectAdBlock} from "adblock-detect-react";
+import {usePreventScrolling} from "../../../organisms/ad-block-overlay/preventScrolling";
+import {getJWT} from "../../../services/AuthService";
+import config from "../../../config/config";
+import {PostModel} from "./PostModel";
+import {Post} from "./Post";
+import {User} from "./User";
+
+
+interface UserModel {
```
Comment (maxschwinghammer) [inline src/main/web/src/scenes/Search/components/SearchService.tsx:14]: Best practise: Extract this to a seperate file called `UserModel` in a folder `models`

File: src/main/web/src/scenes/Search/index.tsx [hunk 4]
```diff
@@ -1,108 +1,50 @@
...
+        <div className="search-content">
         <div className="search-sidebar">
           <SearchSortOptions onSortChange={handleSortChange}/>
           <FindByOptions onSortChange={handleFindByChange}/>
+        </div>
+            <div className="search-result-container">
+                <SearchService sortOption={sortOption} query={searchTerm} findByOption={findOption}/>
+            </div>
         </div>
         <Footer/>
```
Comment (maxschwinghammer) [inline src/main/web/src/scenes/Search/index.tsx:49]: Please add mobile footer (see other pages for implementation)

File: src/main/web/src/scenes/User/index.tsx [hunk 5]
```diff
@@ -1,96 +1,115 @@
...
+        ...config.headers,
+        'Authorization': jwt ? `Bearer ${jwt}` : ''
+    };
 
-  const adBlockDetected: boolean = useDetectAdBlock();
-  usePreventScrolling(adBlockDetected);
+    useEffect((): void => {
+        const fetchUserData = async (): Promise<void> => {
+            try {
+                const response: Response = await fetch(`https://localhost:8443/user/user-information/${id}`, {
```
Comment (maxschwinghammer) [inline src/main/web/src/scenes/User/index.tsx:30]: Do not use localhost, change it to `config.apiUrl`

File: src/main/web/src/atoms/Pictures/Pictures.tsx
Comment (maxschwinghammer) [inline src/main/web/src/atoms/Pictures/Pictures.tsx:1]: Why a random picture instead of the default profile picture?

**Review by maxschwinghammer — CHANGES_REQUESTED** [review]
Added some comments

**Review by maxschwinghammer — APPROVED** [review]

**Review by denniskp — APPROVED** [review]

**Pull request merged**